x-pack/legacy/plugins/index_management/public/services/api.ts

```diff
@@ -201,19 +214,66 @@ export function loadIndexTemplates() {
   });
 }
 
-export const deleteTemplates = async (names: Array<Template['name']>) => {
-  const uimActionType = names.length > 1 ? UIM_TEMPLATE_DELETE_MANY : UIM_TEMPLATE_DELETE;
-
-  return sendRequest({
+export async function deleteTemplates(names: Array<Template['name']>) {
+  const result = sendRequest({
     path: `${apiPrefix}/templates/${names.map(name => encodeURIComponent(name)).join(',')}`,
     method: 'delete',
-    uimActionType,
   });
-};
+
+  const uimActionType = names.length > 1 ? UIM_TEMPLATE_DELETE_MANY : UIM_TEMPLATE_DELETE;
+
+  trackUiMetric(METRIC_TYPE.COUNT, uimActionType);
+
+  return result;
+}
 
 export function loadIndexTemplate(name: Template['name']) {
   return useRequest({
     path: `${apiPrefix}/templates/${encodeURIComponent(name)}`,
     method: 'get',
   });
 }
+
+export async function loadIndexPatterns() {
+  const { savedObjects } = await getSavedObjectsClient().find({
+    type: 'index-pattern',
+    fields: ['title'],
+    perPage: 10000,
+  });
+
+  return savedObjects;
+}
+
+export async function saveTemplate(template: Template, isClone: boolean) {
+  const result = sendRequest({
+    path: `${apiPrefix}/templates`,
+    method: 'put',
+    body: template,
+  });
+
+  const uimActionType = isClone ? UIM_TEMPLATE_CLONE : UIM_TEMPLATE_CREATE;
+
+  trackUiMetric(METRIC_TYPE.COUNT, uimActionType);
+
+  return result;
+}
+
+export async function updateTemplate(template: Template) {
+  const { name } = template;
+  const result = sendRequest({
+    path: `${apiPrefix}/templates/${encodeURIComponent(name)}`,
+    method: 'put',
+    body: template,
+  });
+
+  trackUiMetric(METRIC_TYPE.COUNT, UIM_TEMPLATE_UPDATE);
+
+  return result;
+}
+
+export async function loadTemplateToClone(name: Template['name']) {
```

Comment:
Nit: Why not call it simply `loadTemplate()`. As it could be used in a different context (not only cloning).